mc: 4.8.23 -> 4.8.24

Izorkin opened the pull request:
###### Motivation for this change
Updated to version 4.8.24

Fixed warning
```./configure: line 10014: /usr/bin/file: No such file or directory```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Izorkin committed:
mc: 4.8.23 -> 4.8.24

lordcirth commented:
closure sizes:
master  4.8.23  115.3M
pr 4.8.24 115.4M

All binaries build and run on my workstation (Ubuntu 18.04)

Minor version bump, LGTM.

Izorkin commented:
cc @Mic92

Izorkin commented:
сс @alyssais @Mic92 @FRidh @xeji

Mic92 committed:
mc: add autoreconfHook instead of /usr/bin/file patch

The error is still there but harmless.
It should be rather fixed in autoconf than in every package that
uses autotools.

Mic92 committed:
mc: quote urls
